kivy_garden/mapview/utils.py

```diff
@@ -43,8 +44,191 @@ def get_zoom_for_radius(radius_km, lat=None, tile_size=256.0):
+    with suppress(OverflowError):
+        while (
+            earth_circumference
+            / (2 << (zoom - 1)) * nr_tiles_shown
+            > 2 * radius
+        ):
+            zoom += 1
     return zoom - 1  # Go one zoom level back
+
+
+def get_bounding_box(locations):
+    """
+    Calculate the minimum and maximum latitude and longitude
+    from the given set of coordinates to form a bounding box
+
+    :Parameters:
+        `locations`: List of tuples containing latitude and longitude.
+    """
+    min_lat = min(locations, key=lambda x: x[0])[0]
+    max_lat = max(locations, key=lambda x: x[0])[0]
+    min_lon = min(locations, key=lambda x: x[1])[1]
+    max_lon = max(locations, key=lambda x: x[1])[1]
+    return min_lat, max_lat, min_lon, max_lon
+
+
+def get_bounding_box_center(locations):
+    """
+    Find the center of this bounding box by averaging the
+    minimum and maximum latitudes and longitudes
+
+    :Parameters:
+        `locations`: List of tuples containing latitude and longitude.
+    """
+    min_lat, max_lat, min_lon, max_lon = get_bounding_box(locations)
+    center_lat = (min_lat + max_lat) / 2
+    center_lon = (min_lon + max_lon) / 2
+    return center_lat, center_lon
+
+
+def get_fit_zoom_level(locations, map_width, map_height):
+    """
+    Calculates the zoom level to fit all locations into the map view.
+
+    Determine the zoom level that fits the bounding box within the map view.
+    This involves calculating the required scale to fit both the width
+    and height of the bounding box into the viewport.
+
+    :Parameters:
+        `locations`: List of tuples containing latitude and longitude.
+        `map_width`: Width of the map
+        `map_height`: Height of the map
+
+    :return: Calculated zoom level.
+    """
+    min_lat, max_lat, min_lon, max_lon = get_bounding_box(locations)
+
+    # map_width /= 5
+    # map_height /= 5
+
+    # Constants for zoom calculation
+    TILE_SIZE = 256
+
+    # Function to convert latitude to pixel value
+    def lat_to_pixel(lat, zoom):
+        return (
+            TILE_SIZE
+            * (1 - log(tan(radians(lat)) + 1 / cos(radians(lat))) / pi)
+            / 2 * (2 ** zoom)
+        )
+
+    # Function to convert longitude to pixel value
+    def lon_to_pixel(lon, zoom):
+        return TILE_SIZE * (lon + 180) / 360 * (2 ** zoom)
+
+    # Determine the best zoom level
+    zoom = 1
+    for z in range(1, 21):  # Assuming a max zoom level of 20
+        lat_pixel_range = lat_to_pixel(max_lat, z) - lat_to_pixel(min_lat, z)
+        lon_pixel_range = lon_to_pixel(max_lon, z) - lon_to_pixel(min_lon, z)
+
+        if lat_pixel_range < map_height and lon_pixel_range < map_width:
+            zoom = z
+        else:
+            break
+
+    return zoom
+
+
+def update_map_view(
+        map_width,
+        map_height,
+        lat1,
+        lon1,
+        lat2,
+        lon2,
+        mapview=None,
+        polyline_layer=None,
+        max_zoom=16
+
```

Comment:
Same here, please pass the tile_size as well, and pass it to the function that accept it